src/framework/audio/engine/internal/export/soundtrackwriter.cpp

```
@@ -104,13 +104,17 @@ Ret SoundTrackWriter::write()
         if (!m_isAborted) {
             m_encoderPtr->end();
         }
+        m_isAborted = false;
 
-        audioEngine()->setMode(RenderMode::IdleMode);
-
-        m_source->setOutputSpec(audioEngine()->outputSpec());
-        m_source->setIsActive(false);
+        //! NOTE Changes to the source and audio engine state
+        // must be performed via execOperation - so that synchronization with the audio driver process works
+        IAudioEngine::Operation func = [this]() {
```

Cppcheck (2.20.0)

[style] The function 'setInputParams' is never used.

(unusedFunction)